features/steps/component.go

```diff
@@ -2,235 +2,101 @@ package steps
 
 import (
 	"context"
-	"errors"
-	"fmt"
 	"net/http"
-	"os"
-	"os/signal"
-	"sync"
-	"syscall"
+	"strconv"
 	"testing"
 	"time"
 
-	"github.com/maxcnunes/httpfake"
-
 	"github.com/ONSdigital/dp-cantabular-filter-flex-api/config"
-	"github.com/ONSdigital/dp-cantabular-filter-flex-api/datastore/mongodb"
 	"github.com/ONSdigital/dp-cantabular-filter-flex-api/features/mock"
-	"github.com/ONSdigital/dp-cantabular-filter-flex-api/model"
 	"github.com/ONSdigital/dp-cantabular-filter-flex-api/service"
-	servicemock "github.com/ONSdigital/dp-cantabular-filter-flex-api/service/mock"
 	componenttest "github.com/ONSdigital/dp-component-test"
-	"github.com/ONSdigital/dp-component-test/utils"
-	kafka "github.com/ONSdigital/dp-kafka/v3"
 	"github.com/ONSdigital/log.go/v2/log"
 )
 
 const (
 	ComponentTestGroup    = "test-consumer-group"
 	DrainTopicTimeout     = 10 * time.Second // maximum time to wait for a topic to be drained
 	DrainTopicMaxMessages = 1000             // maximum number of messages that will be drained from a topic
-	MinioCheckRetries     = 3                // maximum number of retires to validate that a file is present in minio
 	WaitEventTimeout      = 10 * time.Second // maximum time that the component test consumer will wait for a
 )
 
 var (
-	BuildTime string = "1625046891"
-	GitCommit string = "7434fe334d9f51b7239f978094ea29d10ac33b16"
-	Version   string = ""
+	BuildTime = strconv.Itoa(time.Now().Nanosecond())
+	GitCommit = "component test commit"
+	Version   = "component test version"
 )
 
 type Component struct {
 	componenttest.ErrorFeature
-	ApiFeature        *componenttest.APIFeature
-	errorChan         chan error
-	DatasetAPI        *httpfake.HTTPFake
-	CantabularClient  *mock.CantabularClient
-	svc               *service.Service
-	cfg               *config.Config
-	wg                *sync.WaitGroup
-	signals           chan os.Signal
-	ctx               context.Context
-	HTTPServer        *http.Server
-	store             service.Datastore
-	g                 service.Generator
-	shutdownInitiated bool
-	consumer          kafka.IConsumerGroup
-	waitEventTimeout  time.Duration
+	AuthServiceInjector *componenttest.AuthorizationFeature
```

Comment:
I would really prefer we keep these named features rather than injectors, as it's what they're called in the packages and we're all used to that. I don't think you'll find anyone that finds it makes it considerably clearer what they do.

Reply:
yeah I don't  have an issue with that